src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -651,8 +654,8 @@ end
   cf_rr = calc_fast_wavespeed(u_rr, orientation, equations)
 
   # Calculate velocities
-  v_ll = zero(u_ll[1])
-  v_rr = zero(u_rr[1])
+  v_ll = 0
+  v_rr = 0
```

Comment:
```suggestion
  v_ll = zero(u_ll[1])
  v_rr = zero(u_rr[1])
```
Otherwise type unstable